src/upload-artifact.ts

```diff
@@ -0,0 +1,43 @@
+import * as core from '@actions/core'
+import * as artifact from '@actions/artifact'
+import {Inputs, getDefaultArtifactName} from './constants'
+import {findFilesToUpload} from './search'
+
+async function run(): Promise<void> {
+  try {
+    const name = core.getInput(Inputs.Name, {required: false})
+    const path = core.getInput(Inputs.Path, {required: true})
+
+    const searchResult = await findFilesToUpload(path)
+    if (searchResult.filesToUpload.length === 0) {
+      core.warning(
+        `No files were found for the provided path: ${path}. No artifacts will be uploaded.`
+      )
+    } else {
+      core.info(
+        `With the provided path, there will be ${searchResult.filesToUpload.length} files uploaded`
+      )
+      for (const file of searchResult.filesToUpload) {
```

Comment:
Do we need to loop through this? Does the artifact package itself have logging that we can use instead?

When debug logging is turned off, this will just use up time.

Reply:
The package itself does not which is why I added this. It felt very barren. I will be quickly adding some changes/fixes to the package so I can add some extra logging there and get rid of this loop here.

![image](https://user-images.githubusercontent.com/16109154/75054815-ca605480-54a1-11ea-9e65-ca01f4854d5c.png)

Reply:
I think ideally we would log as we upload each file, so users have a better idea of the progress made. If we have to do that in the package itself, it might make sense to add a `verbose` option so that it's not logged by default.

Reply:
Good news, I've overhauled the logging in `@actions/artifact` with `v0.1.1`, offers way more information compared to `v0.1.0`

The behavior is much like `v1` but just with some small modifications:
![image](https://user-images.githubusercontent.com/16109154/76530558-2bad8f00-6474-11ea-9ebb-18f49d9293cf.png)

- Every 10 seconds the total upload progress will be displayed
- If an individual file is over 100MB in size, it will display individual information about the status of it's upload every second

With debug on, the total time for an individual file to upload also gets displayed
![image](https://user-images.githubusercontent.com/16109154/76532027-54cf1f00-6476-11ea-8bba-eebb5407a218.png)